src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
On 09/04/15 22:38, Matt Broadstone wrote:
> In src/qamqpclient.cpp
> <https://github.com/mbroadst/qamqp/pull/26#discussion_r28057490>:
> 
>> @@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
>>  QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
>>  {
>>      Q_D(QAmqpClient);
>> +    if (!name.isEmpty() && d->exchanges.contains(name)
>> +                    && (!d->exchanges[name].isNull()))
> 
> what if |d->exchanges[name].isNull()|? This hash continues to grow even
> though exchanges might be deleted right?
> 

It will, but only by very small increments.  The alternative is I hook
the 'deleted' signal from queues and exchanges, to trigger a clean up of
both lists.

I don't do a remove here, since if the pointer is indeed NULL, it soon
won't be.
-- 
Stuart Longland (aka Redhatter, VK4MSL)

I haven't lost my mind...
  ...it's backed up on a tape somewhere.